karaf/init/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/osgi/ToscaTypePlanTransformerClassloaderWrapper.java

```diff
@@ -0,0 +1,126 @@
+package io.cloudsoft.tosca.a4c.brooklyn.osgi;
+
+import io.cloudsoft.tosca.a4c.brooklyn.plan.ToscaTypePlanTransformer;
+import org.apache.brooklyn.api.mgmt.ManagementContext;
+import org.apache.brooklyn.api.typereg.RegisteredType;
+import org.apache.brooklyn.api.typereg.RegisteredTypeLoadingContext;
+import org.apache.brooklyn.core.typereg.BrooklynTypePlanTransformer;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * The purpose of this class is to set the thread context classloader upon all calls to
+ * a delegate {@link ToscaTypePlanTransformer}.  This is done so that Spring lookups
+ * using org.springframework.util.ClassUtils#getDefaultClassLoader() will find the
+ * classloader for this bundle, rather than whatever may happen to be in place at the
+ * point of call.
+ */
+public class ToscaTypePlanTransformerClassloaderWrapper implements BrooklynTypePlanTransformer {
+
+    ToscaTypePlanTransformer delegateTransformer;
+
+    public ToscaTypePlanTransformerClassloaderWrapper(ToscaTypePlanTransformer toscaTypePlanTransformer) {
+        final ClassLoader original = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(toscaClassLoader());
```

Comment:
Presumably setting the contextClassLoader just before setting the field doesn't do anything? But we include the code so that the same pattern is followed in every method/constructor? I'm happy with that.